### PR TITLE
Use private interface if there's no public_vlan defined

### PR DESCRIPTION
**Pull request by gczuczy**
This is for issue https://github.com/saltstack/salt/issues/19948
Please review and test it, should be good.

### CODE REVIEW

File: salt/cloud/clouds/softlayer.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def create(vm_):
     private_wds = config.get_cloud_config_value(
         'private_windows', vm_, __opts__, default=False
     )
-    if private_ssh or private_wds:
+    if private_ssh or private_wds or public_vlan == None or public_vlan == False:
```
Comment (techhat) [inline salt/cloud/clouds/softlayer.py]: `is None` and `is False`

**Comment by cachedout** [comment]
@gczuczy Please let us know when you're able to make the above changes and we'll get this merged. Thanks very much!

**Comment by gczuczy** [comment]
Should be there, please check.

**Comment by cachedout** [comment]
Got it! Thanks, @gczuczy

**Pull request merged**